Mathlib/Order/RelSeries.lean

```diff
@@ -211,18 +213,59 @@ def append (p q : RelSeries r) (connect : r p.last q.head) : RelSeries r where
         Nat.add_assoc _ 1, add_comm 1, Nat.sub_add_cancel]
       exact hi
 
+lemma append_apply_left (p q : RelSeries r) (connect : r p.last q.head)
+    (i : Fin (p.length + 1)) :
+    p.append q connect i = p i := by
```

Comment:
Can you use `addNat` rather than the poorly behaved coercion `Fin a → Fin b` which is the composition of `Fin a → ℕ` and `ℕ → Fin b`?